securemailbox/views/retrieve.py

```diff
@@ -1,11 +1,78 @@
-from flask import Blueprint, jsonify
+from flask import Blueprint, jsonify, request, json
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.orm.exc import NoResultFound
+
+from ..models import Mailbox
+from ..models import Message
 
 from securemailbox import db
 
 # Create the `retrieve` blueprint
 retrieve_blueprint = Blueprint("retrieve", __name__)
 
-
 @retrieve_blueprint.route("/retrieve/", methods=["POST"])
 def retrieve():
-    return jsonify({"success": True, "error": None})
+
+    # Check for valid json request
+    if not request.is_json:
+        return jsonify({'success': False, 'error': 'Request must be valid json', 'data': None}), 400
+
+    req_fingerprint = request.json.get('fingerprint', None)
+
+    # Check if fingerprint is present in json request body
+    if req_fingerprint is None:
+        return jsonify({'success': False, 'error': 'fingerprint not in request', 'data': None}), 400
+
+    # Check if fingerprint value is string variable
+    if not isinstance(req_fingerprint, str):
+        return jsonify({'success': False, 'error': 'fingerprint value not valid', 'data': None}), 400
+
+    # search for fingerprint in mailbox table
+    try:
+        mailbox_id = Mailbox.query.filter_by(fingerprint=req_fingerprint).first()
```

Comment:
Get just the id here so you don't have to do `mailbox_id = mailbox_id.id` later. I'm not sure the best way to do so but you could do `mailbox_id = (Mailbox.query.filter_by(fingerprint=req_fingerprint).first()).id` which works but I'm sure is not the "right" way to do it.

Reply:
I figured out how to do this, by just adding .id at the end of the query, but it fails when the fingerprint is not in the database, because it's trying to get the .id attribute from a None type object. So I think I have to first just get the object returned and check if it's None before setting it to the id field.

Reply:
I think you can do something like this:
```python
mailbox_id = Mailbox.query(id).filter_by(fingerprint=req_fingerprint).first()
```

Docs on SQLAlchemy querying are [here](https://docs.sqlalchemy.org/en/13/orm/query.html).